data/transactions/logic/box.go

```diff
@@ -165,9 +169,60 @@ func MakeBoxKey(appIdx basics.AppIndex, name string) string {
 	   The "bx:" prefix is so that the kvstore might be usable for things
 	   besides boxes.
 	*/
-	key := make([]byte, 3 /* bx: */ +8 /* appIdx, big-endian */ +len(name))
-	copy(key, "bx:")
-	binary.BigEndian.PutUint64(key[3:], uint64(appIdx))
-	copy(key[11:], name)
+	key := make([]byte, boxNameIndex + len(name))
+	copy(key, boxPrefix)
+	binary.BigEndian.PutUint64(key[boxPrefixLength:], uint64(appIdx))
+	copy(key[boxNameIndex:], name)
 	return string(key)
 }
+
+// GetAppAndNameFromKey extracts an appid and box name from a string that was created by MakeBoxKey()
+func GetAppAndNameFromKey(key string) (basics.AppIndex, string, error) {
+	// no assertion that prefix is "bx:"
+	if len(key) < boxPrefixLength+8 {
+		return 0, "", fmt.Errorf("GetAppNameFromKey() cannot extract AppIndex as key too short (length=%d)", len(key))
+	}
+	// WARNING: even junk such as "this is definitely NOT a box key" will return without error
```

Comment:
To gain a sliver of speed, am dispensing with the check that `key[:boxPrefixLength] == boxPrefix`